src/Config/Processor/BuilderProcessor.php

```diff
@@ -126,30 +151,92 @@ private static function processFieldBuilders(array $fields)
             }
 
             if ($fieldBuilderName) {
-                $buildField = self::getBuilder($fieldBuilderName, self::BUILDER_FIELD_TYPE)->toMappingDefinition($builderConfig);
-                $field = \is_array($field) ? \array_merge($buildField, $field) : $buildField;
+                $mapping = self::getFieldBuilderMapping($fieldBuilderName, self::BUILDER_FIELD_TYPE, $builderConfig, $reservedTypesMap);
+
+                $fieldMapping = $mapping['field'];
+                $field = \is_array($field) ? \array_merge($fieldMapping, $field) : $fieldMapping;
+                $newTypes = \array_merge($newTypes, $mapping['types']);
             }
             if (isset($field['argsBuilder'])) {
                 $field = self::processFieldArgumentsBuilders($field);
             }
         }
 
-        return $fields;
+        return [
+            'fields' => $fields,
+            'types' => $newTypes,
+        ];
     }
 
-    private static function processFieldsBuilders(array $builders)
+    private static function processFieldsBuilders(array $builders, array &$reservedTypesMap)
     {
         $fields = [];
+        $newTypes = [];
+
         foreach ($builders as $builder) {
             $builderName = $builder['builder'];
-            $builderConfig = null;
-            if (isset($builder['builderConfig'])) {
-                $builderConfig = $builder['builderConfig'];
+            $builderConfig = $builder['builderConfig'] ?? null;
+
+            $mapping = self::getFieldBuilderMapping($builderName, self::BUILDER_FIELDS_TYPE, $builderConfig, $reservedTypesMap);
```

Comment:
Scrutinizer
/ Inspection => It seems like ``$builderConfig`` can also be of type ``null``; however, parameter ``$builderConfig`` of ``Overblog\GraphQLBundle\Config\Processor\BuilderProcessor::getFieldBuilderMapping()`` does only seem to accept ``array``, maybe add an additional type check?

Reply:
This is right as you say it. I also seen this. The problem was, before it was by default null'ed too and passed to the MappingInterface API which also only accepted array (not nullable). So I was thinking fixing this should be BC because the code that had the null case would not have worked for anybody anyway. Since it was an error unrelated to this PR I also did not want to introduce new exception, as this might be seen as BC break (before no exception, error). Also defaulting it to empty array is BC break I realized, because before it failed with error, but now it would silently pass empty config. So you want type check and exception?

Reply:
ok I'll fix from 0.11 it after merging this